### DISCUSSION TITLE
How does the Reset method works in Popup ?

### DISCUSSION THREAD

**Discussion by mosflex** [xamarin/XamarinCommunityToolkit#1933] [Q&A] [answered]
Hi,
I am working with popup and I am interested by the Reset method. In the documentation we can read the description : **_"Resets the Popup so that another Result can be returned."_** but I don't understand how can another result can be returned if we don't "reawait" the result ? This method is not awaitable so how can we use it. I don't find an example.  Thanks in advance !

**Comment by mosflex** [comment, marked as answer]
Ok I understood now when I tried to call ShowPopupAsync() in a loop.  
When the popup is dismissed first time and recalled second time it keep always the same result. 
So to have another result, you have to reset the popup before calling ShowPopupAsync() second time

**Reply by bijington** [reply]
Yes that is correct `Reset` needs to be called if you wish to show the `Popup` another time and await its result
